compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1033,3 +1039,371 @@ OMR::ARM64::TreeEvaluator::generateVFTMaskInstruction(TR::CodeGenerator *cg, TR:
    // Do nothing in OMR
    return preced;
    }
+
+/**
+ * @brief Inlines an intrinsic for calls to atomicAddSymbol which are represented by a call node of the form for 32-bit (64-bit similar):
+ *
+ *     This implies `volatile` memory access mode.
+ *
+ *     @code
+ *       icall <atomicAddSymbol>
+ *         <address>
+ *         <value>
+ *     @endcode
+ *
+ *     Which performs the following operation atomically:
+ *
+ *     @code
+ *       [address] = [address] + <value>
+ *       return <value>
+ *     @endcode
+ *
+ * @param node: The respective (i|l)call node.
+ * @param   cg: The code generator used to generate the instructions.
+ * @returns A register holding the <value> node.
+ */
+static TR::Register* intrinsicAtomicAdd(TR::Node* node, TR::CodeGenerator* cg)
+   {
+   TR::Node* addressNode = node->getChild(0);
+   TR::Node* valueNode = node->getChild(1);
+
+   TR::Register* addressReg = cg->evaluate(addressNode);
+   TR::Register* valueReg = cg->evaluate(valueNode);
+   const bool is64Bit = valueNode->getDataType().isInt64();
+
+   TR::Register *oldValueReg = cg->allocateRegister();
+
+   /*
+    * Generating non-intuitive instruction sequence which uses load exclusive register 
+    * and store release exclusive register followed by full memory barrier.
+    *
+    * Because this atomic add has `volatile` semantics,
+    * no loads/stores before this sequence can reordred after it and
+    * no loads/stores after it can reordered before it.
+    *
+    * loop:
+    *    ldxrx   oldValue, [addressReg]
```

Comment:
`oldValue` here should be `oldValueReg`.

Reply:
fixed.